drt_1d.py

```diff
@@ -88,10 +195,50 @@ def construct_transition_matrix(num_voxels:int, voxel_size:float, sigma_s:np.nda
     for i in range(num_voxels):
         for j in range(num_voxels):
             if i == j:
-                # Set diagonal values to one based on defintion of streaming operator
-                K_trans[i, j] = 1
+                # Diagonal element (same voxel)
+                # Retrieve the list of positions for the current voxel i (same as j in this case)
+                positions_in_voxel = positions.get(str(i))
+                
+                # Check if there are multiple positions within this voxel
+                if len(positions_in_voxel) > 1:
+                    # Calculate the total number of position pairs (combinations of 2)
+                    position_pairs = itertools.combinations(positions_in_voxel, 2)
+                    
+                    streaming_operator_sum = 0.0
+                    num_position_pairs = 0
+                    for position_a, position_b in position_pairs:
+                        # Calculate the streaming operator for the pair (both directions)
+                        streaming_operator_sum += calculate_K_trans(i, j, position_a, position_b, voxel_size, sigma_s)
+                        streaming_operator_sum += calculate_K_trans(i, j, position_b, position_a, voxel_size, sigma_s)
+                        num_position_pairs += 2
```

Comment:
1. streaming operator is symmetric, don't need to go both ways
2. `num_position_pairs` will always be `2 * len(position_pairs)` so why both accumulating